js/mixins/multipe-lines.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * (c) 2010-2018 Wojciech Chmiel
+ *
+ * License: www.highcharts.com/license
+ */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+import '../parts/Utilities.js';
+
+var each = H.each,
+    merge = H.merge,
+    error = H.error,
+    defined = H.defined,
+    SMA = H.seriesTypes.sma;
+
+/**
+ * Mixin useful for all indicators that have more than one line.
+ * Merge it with your implementation where you will provide
+ * getValues method appropriate to your indicator and pointArrayMap,
+ * pointValKey, linesApiNames properites. Notice that pointArrayMap
+ * should be consistent with amount of lines calculated in getValues method.
+ */
+var multipleLinesMixin = {
+    /**
+     * Lines ids. Required to plot appropriate amount of lines.
+     * Notice that pointArrayMap should have more elements than
+     * linesApiNames, because it containes main line and additional lines ids.
+     * Also it should be consistent with amount of lines calculated in
+     * getValues method from your implementation.
+     * @type {Array}
+     */
+    pointArrayMap: ['top', 'bottom'],
+
+    /**
+     * Main line id.
+     * @type {string}
+     */
+    pointValKey: 'top',
+
+    /**
+     * Additional lines DOCS names. Elements of linesApiNames array should
+     * be consistent with DOCS line names defined in your implementation.
+     * Notice that linesApiNames should have decreased amount of elements
+     * relative to pointArrayMap (without pointValKey).
+     * @type {Array}
+     */
+    linesApiNames: ['bottomLine'],
+
+    /**
+     * Create translatedLines Collection based on pointArrayMap.
+     * @param {string} excludedValue pointValKey - main line id
+     * @returns {Array} Returns translated lines names without excluded value.
+     */
+    getTranslatedLinesNames: function (excludedValue) {
+        var translatedLines = [];
+
+        each(this.pointArrayMap, function (propertyName) {
+            if (propertyName !== excludedValue) {
+                translatedLines.push(
+                    'plot' +
+                    propertyName.charAt(0).toUpperCase() +
```

Comment:
Is not simpler to user `propertyName[0]` instead of  `propertyName.charAt(0)` ?

Reply:
It's building camelcase names: `plotbottomLine` -> `plotBottomLine`